### PR TITLE
Code cleanup and fixed some typos

### PR DESCRIPTION
**Pull request by Jeroenvh99**
I reverted all the changes that weren't linked list related from my last PR, it's ok now.

### CODE REVIEW

File: src/utils/mlx_list.c [hunk 1]
```diff
@@ -18,8 +18,9 @@ int32_t mlx_lstsize(mlx_list_t* lst)
 {
 	int32_t	i = 0;
 
-	if (!lst)
+	/*if (!lst)
 		return (i);
+	this check isn't necessary since the while loop won't execute when lst is NULL*/
```
Comment (W2Wizard) [inline src/utils/mlx_list.c:23]: If thats the case just remove the section of code as well as the comment.

Comment (Jeroenvh99) [reply src/utils/mlx_list.c]: coming up

**Comment by Jeroenvh99** [comment]
I honestly don't know why the build-win test is failing, I didn't actually change any code, so it should just pass that check I'd think

**Comment by Jeroenvh99** [comment]
I looked through all of the documentation and fixed most of the typos I came across

**Comment by Jeroenvh99** [comment]
Good to know, I can look into how to fix that later

**Comment by W2Wizard** [comment]
Great PR! Thanks so much for your time and effort 😄
Checked all the things, i'm happy with this.

**Comment by Jeroenvh99** [comment]
Happy to hear that

**Pull request merged**